src/Map/Map.jsx

```diff
@@ -0,0 +1,63 @@
+import React, {Component, PropTypes} from 'react'
+import ReactHighcharts from 'react-highcharts'
+import {LayerGroup, LayersControl, Map as LeafMap, Marker, Polyline, Popup, TileLayer, ZoomControl} from 'react-leaflet'
+
+export class Map extends Component {
+  static propTypes = {
+    center: PropTypes.array.isRequired,
+    markers: PropTypes.array.isRequired
+  };
+
+  render () {
+    const MyPopupMarker = ({map, position, children}) => (
+     <Marker map={map} position={position}>
+        <Popup>
+          <span>{children}</span>
+        </Popup>
+     </Marker>
+    )
+
+    const MyMarkersList = ({map, markers}) => {
+      const items = markers.map(({ key, ...props }) => (
+        <MyPopupMarker key={key} map={map} {...props} />
+      ));
+      return <div style={{display: 'none'}}>{items}</div>
+    }
+   
+    const baseLayer = (
+      <TileLayer
+        layerName='OpenStreetMap'
+       url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
+       attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+      />
+    )
+   
+    const MyLayerGroup = ({map}) => (
```

Comment:
This shouldn't be defined in the render function. This is a component, so it should be defined outside of the Map component.

Also, just call it `LayerGroup` instead of `MyLayerGroup`